doc/packages/build-support.md

```diff
@@ -0,0 +1,97 @@
+# Build Support {#sec-build-support}
+
+## `pkgs.substitute` {#pkgs-substitute}
+
+`pkgs.substitute` is a wrapper around [the `substitute` function](#fun-substitute) provided by the
+stdenv. It replaces strings in `src` as specified by the `substitutions` argument.
+
+In a build script, the line:
+
+```bash
+substitute $infile $outfile --replace-fail @foo@ ${foopkg}/bin/foo
+```
+
+is equivalent to:
+
+```nix
+{ substitute, foopkg }:
+substitute {
+  src = ./sourcefile.txt;
+  substitutions = [
+    "--replace"
+    "@foo@"
+    "${foopkg}/bin/foo"
+  ];
+}
+```
+
+## `pkgs.substituteAll` {#pkgs-substituteall}
+
+`pkgs.substituteAll` substitutes all instances of `@varName@` (`@`s included)
+in file file `src` with the value of the corresponding environment variable. As
+this uses the `substituteAll` function, its limitations regarding variable names
+that will or will not be replaced also apply here, and familiarity with [the
+function's documentation](#fun-substitute) is recommended.
```

Comment:
```suggestion
this uses the [`substituteAll`] (#fun-substitute) function, its limitations regarding variable names that will or will not be replaced also apply here.
```

Please don't conceal the origin of recommendations with passive voice. Also I recommend against recommendations in a reference manual. We can just link to authoritative information, the reader can expect to find anything they may still be missing there, and we should expect readers to read what they need. If that's not clear enough, in a different effort we may introduce this convention explicitly.

Also, more words is not always more better. Reference material primarily needs to be scannable for what one needs. Lots of prose tends to obscure key information, so we should try to optimize for brevity and clarity.